sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlobAsyncClient.java

```diff
@@ -733,17 +734,31 @@ private <T> Mono<T> populateRequestConditionsAndContext(BlobRequestConditions re
 
                 requestConditionsFinal.setIfMatch(response.getValue().getETag());
                 Mono<T> result = downloadCall.get();
-                if (response.getValue().getMetadata().get(ENCRYPTION_DATA_KEY) != null) {
-                    result = result.contextWrite(context ->
-                        context.put(ENCRYPTION_DATA_KEY,
-                            EncryptionData.getAndValidateEncryptionData(
-                                response.getValue().getMetadata().get(ENCRYPTION_DATA_KEY), requiresEncryption))
-                    );
+
+                String encryptionDataKey = getEncryptionDataKey(response.getValue().getMetadata());
+                if (encryptionDataKey != null) {
+                    result = result.contextWrite(context -> context.put(ENCRYPTION_DATA_KEY,
+                        EncryptionData.getAndValidateEncryptionData(encryptionDataKey, requiresEncryption)));
                 }
                 return result;
             });
     }
 
+    private static String getEncryptionDataKey(Map<String, String> metadata) {
```

Comment:
One more nit, this is repeated multiple times in the same package, let's make this a shared utility method so we don't have skew on the implementations over time.